aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -172,28 +201,45 @@ SparseTensor& asin_sparse_(SparseTensor& t) {
 
 // TODO: add in-place variant
 
-SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t_) {
-  TORCH_CHECK(r.is_sparse(), "Tensor should be sparse");
-  TORCH_CHECK(t_.is_sparse(), "Tensor should be sparse");
-
-  // This coalesce is why we can't easily provide an inplace variant
-  SparseTensor t = t_.coalesce();
-
-  r.resize_as_(t);
-  auto indices = r._indices();
-  indices.resize_as_(t._indices());
-  indices.copy_(t._indices());
-  Tensor r_values = r._values();
-  at::sqrt_out(r_values, t._values());
-  get_sparse_impl(r)->set_nnz_and_narrow(t._nnz());
-  return r._coalesced_(t.is_coalesced());
+SparseTensor& sqrt_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  sparse_unary_op_check_sanity(r, t, true);
+  r._values().sqrt_();
+  return r;
 }
 
 SparseTensor sqrt_sparse(const SparseTensor& t) {
-  SparseTensor r = get_result_tensor_for_unary_op(t);
+  SparseTensor r = get_result_tensor_for_unary_op(t, true);
   sqrt_out_sparse(r, t);
   return r;
 }
+
+SparseTensor& sqrt_sparse_(SparseTensor& t) {
+  return sqrt_out_sparse(t, t);
+}
+
+// --------------------------------------------------------------------
+// asinh(SparseTensor)
+// --------------------------------------------------------------------
+
+// In-place asinh on uncoalesced tensors is not supported since the operation is
+// not a linear map. Values of uncoalesced tensor corresponding to the same
+// indices are summed and asinh(summed_value) != asinh(v1) + asinh(v2)
+
+SparseTensor& asinh_out_sparse(SparseTensor& r, const SparseTensor& t) {
+  sparse_unary_op_check_sanity(r, t, true);
```

Comment:
change `true` -> `false`, see above.

Reply:
This true is for promotes_integer_to_float; not for is_linear_map; have updated with a comment for greater clarity